proposals/0000-numa-aware-work-stealing-scheduler.md

```diff
@@ -0,0 +1,57 @@
+# NUMA-Aware Work Stealing Scheduler
+
+* Proposal: [SE-0000](0000-template.md)
+* Author: [Sebastian Woelke](https://github.com/aufgang001)
+* Status: Awaiting
+
+## Introduction
+
+On modern multiprocessor systems an actor scheduler has to be aware of the underlying hardware in order to minimize performance penalties.
+
+## Motivation
+
+Non-uniform memory access (NUMA) [1] is a computer architecture for multiprocessor systems.
+A modern server-class system consists of multiple processor chips which are connected via a high-speed interconnect.
+A chip together with its locally attached memory is called a NUMA-node and can access its local memory and the memory of other nodes (remote memory) in a transparent way for the programmer.
+NUMA architecture is the successor to the symmetric multiprocessor (SMP) architecture and scales better for a high number of processors but leads to non-uniform access times between local and remote memory.
+In detail, accessing remote memory will take on newer systems about 30% longer than accessing local memory [2].
+This time increases further when the memory controller is overloaded [2] and when the memory access on a more complex NUMA architectures requires multiple hops.
+To minimize performance penalties on NUMA systems an actor scheduler has to be aware of the underlying hardware architecture and should avoid moving actors between NUMA-nodes.
+
+## Proposed Solution
+
+We propose to extend the current work stealing scheduler with an awareness of the underlying hardware.
+In preparation, we pin each worker thread to a specific core and gather information about hardware topology with the Portable Hardware Locality library (hwloc) [3].
+Pinning workers prevents the operation system from reschedule the threads to other cores and gives CAF runtime full control over its worker.
```

Comment:
> operation system from reschedule
Should this be something like "the operating system from rescheduling"?

> gives CAF runtime
Think, this should be either "gives the CAF runtime environment" or simply "gives CAF".